dev/doc/critical-bugs

```diff
@@ -0,0 +1,221 @@
+  GH issue number: #4157
+  risk: 
+
+  component: "virtual machine" (compilation to bytecode ran by a C-interpreter)
+  summary: wrong universe constraints
+  introduced: possibly exploitable from V8.1; exploitable at least from V8.5
+  impacted released versions: V8.1-V8.4pl5 unknown, V8.5-V8.5pl3, V8.6-V8.6.1, V8.7.0-V8.7.1
+  impacted development branches: unknown for v8.1-v8.4, none from v8.5
+  impacted coqchk versions: none (no virtual machine in coqchk)
+  fixed in: master (c9f3a6cbe5c41, 12 Feb 2018, PR#6713, Dénès), v8.7 (c058a4182, 15 Feb 2018, Zimmermann, backport), v8.6 (a2cc54c6, 21 Feb 2018, Herbelin, backport), v8.5 (d4d550d0f1a, 21 Feb 2018, Herbelin, backport)
+  found by: Dénès
+  exploit: test-suite/bugs/closed/6677.v
+  GH issue number: #6677
+  risk: 
+
+  component: "virtual machine" (compilation to bytecode ran by a C-interpreter)
+  summary: missing pops in executing 31bit arithmetic
+  introduced: V8.5
+  impacted released versions: V8.1-V8.4pl5
+  impacted development branches: v8.1 (probably)
+  impacted coqchk versions: none (no virtual machine in coqchk)
+  fixed in: master/trunk/v8.5 (a5e04d9dd, 6 Sep 2015, Dénès), v8.4 (d5aa3bf6, 9 Sep 2015, Dénès), v8.3 (5da5d751, 9 Sep 2015, Dénès), v8.2 (369e82d2, 9 Sep 2015, Dénès), 
+  found by: Catalin Hritcu
+  exploit: lost?
+  GH issue number: ?
+  risk: 
+
+  component: "native" conversion machine (translation to OCaml which compiles to native code)
+  summary: translation of identifier from Coq to OCaml was not bijective, leading to identify True and False
+  introduced: V8.5
+  impacted released versions: V8.5-V8.5pl1
+  impacted development branches: none
+  impacted coqchk versions: none (no native computation in coqchk)
+  fixed in: master/trunk/v8.6 (a91d15dcb4, 19 May 2016, letouzey), v8.5 (088b3161c93, 19 May 2016, letouzey), 
+  found by: Letouzey, Dénès
+  exploit: lost?
+  GH issue number: ?
+  risk: 
+
+Conflicts with axioms in library
+
+  component: library of real numbers
+  summary: axiom of description and decidability of equality on real numbers in library Reals was inconsistent with impredicative Set
+  introduced: 67c75fa01ad, 20 Jun 2002
+  impacted released versions: 7.3.1, 7.4
+  impacted coqchk versions: 
+  fixed by deciding to drop impredicativity of Set: bac7079739, 28 Oct 2004
+  found by: Herbelin & Werner
+  exploit: need to find the example again
+  GH issue number: no
+  risk: unlikely to be exploited by chance
+
+  component: library of extensional sets, guard condition
+  summary: guard condition was unknown to be inconsistent with propositional extensionality in library Sets
+  introduced: not a bug per se but an incompatibility discovered late
+  impacted released versions: technically speaking from V6.1 with the introduction of the Sets library which was then inconsistent from the very beginning without we knew it
+  impacted coqchk versions: ?
+  fixed by constraining the guard condition: bac7079739, 28 Oct 2004, Barras, Dénès
```

Comment:
bac7079739 is 
> Set devient predicatif par defaut
>
> git-svn-id: svn+ssh://scm.gforge.inria.fr/svn/coq/trunk@4726 85f007b7-540e-0410-9357-904b9bb8a0f7

by @herbelin committed on Oct 28, 2003...

Also, is this the one whose proof is
```coq
Axiom ext : (False -> False) = True.
Fixpoint oops (x : True) : False
  := match x with
     | I => oops (match ext in (_ = T) return T with
                  | eq_refl => fun absurd => match absurd with end
                  end)
     end.
```
which is still present as late as 8.4pl6?

Reply:
Bad copy-paste. I changed for a more relevant commit, but I'm not sure it was fixed by only  one commit and if one commit exactly this one. I'm unaware of the exact history and if @maximedenes could give the needed clarifications and fill the missing bits, that would be wonderful.